DSC/dsc.py

```diff
@@ -344,11 +402,30 @@ def update():
         hutil.do_exit(1, 'Update', 'error', '1', 'Update failed: {0}'.format(e))
 
 def run_cmd(cmd):
-    proc = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE)
-    proc.wait()
-    output = proc.stdout.read()
-    code = proc.returncode
-    return code,output        
+    proc = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True, close_fds=True)
```

Comment:
Leaving shell=True is unsafe and should be avoided if possible.

Reply:
Thank You. These changes are currently pushed to Prod and we do not see any issues as of now. Will see if we can take the change as part of future changes as it requires testing.